### ISSUE TITLE
Ansible connection is sometimes dropped mid-play on windows hosts

### ISSUE THREAD

**Issue by christonog** [ansible/ansible-modules-core#5368]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

win_user.ps1
##### ANSIBLE VERSION

```
ansible 2.1.1.0
  config file = 
  configured module search path = Default w/o overrides
```
##### OS / ENVIRONMENT

I am using the Amazon created flavor of linux. ami id: ami-f5f41398
##### SUMMARY

For the win_user task, ansible fails to reach the remote host, despite already making a connection and performing tasks.
##### STEPS TO REPRODUCE

Can't seem to reproduce this accurately, and it's unfortunately not tied to one specific module (I just so happen to be running the win_user module.

```
TASK [create_users : Create noreply user] **************************************
task path: /home/ec2-user/Ansible/roles/create_users/tasks/main.yml:20
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/ansible/plugins/connection/winrm.py", line 271, in exec_command
    result = self._winrm_exec(cmd_parts[0], cmd_parts[1:], from_exec=True)
  File "/usr/local/lib/python2.7/site-packages/ansible/plugins/connection/winrm.py", line 233, in _winrm_exec
    self.protocol.cleanup_command(self.shell_id, command_id)
  File "/usr/local/lib/python2.7/site-packages/winrm/protocol.py", line 307, in cleanup_command
    res = self.send_message(xmltodict.unparse(req))
  File "/usr/local/lib/python2.7/site-packages/winrm/protocol.py", line 207, in send_message
    return self.transport.send_message(message)
  File "/usr/local/lib/python2.7/site-packages/winrm/transport.py", line 173, in send_message
    response = self.session.send(prepared_request, timeout=self.read_timeout_sec)
  File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 596, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/adapters.py", line 479, in send
    raise ConnectTimeout(e, request=request)
ConnectTimeout: HTTPSConnectionPool(host='172.16.169.179', port=5986): Max retries exceeded with url: /wsman (Caused by ConnectTimeoutError(<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f86cb458f50>, 'Connection to 172.16.169.179 timed out. (connect timeout=30)'))
fatal: [172.16.169.179]: UNREACHABLE! => {"changed": false, "msg": "failed to exec cmd PowerShell -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -EncodedCommand UwBlAHQALQBTAHQAcgBpAGMAdABNAG8AZABlACAALQBWAGUAcgBzAGkAbwBuACAATABhAHQAZQBzAHQACgBUAHIAeQAKAHsACgAmACAAJwBDADoAXABVAHMAZQByAHMAXABhAG4AcwBpAGIAbABlAFwAQQBwAHAARABhAHQAYQBcAEwAbwBjAGEAbABcAFQAZQBtAHAAXABhAG4AcwBpAGIAbABlAC0AdABtAHAALQAxADQANwA3ADMAMQA2ADYAMAAwAC4ANwA3AC0AMQA2ADMAMgAxADgANgA5ADAAOQA2ADUAOQA4ADYAXAB3AGkAbgBfAHUAcwBlAHIALgBwAHMAMQAnAAoAfQAKAEMAYQB0AGMAaAAKAHsACgAkAF8AbwBiAGoAIAA9ACAAQAB7ACAAZgBhAGkAbABlAGQAIAA9ACAAJAB0AHIAdQBlACAAfQAKAEkAZgAgACgAJABfAC4ARQB4AGMAZQBwAHQAaQBvAG4ALgBHAGUAdABUAHkAcABlACkACgB7AAoAJABfAG8AYgBqAC4AQQBkAGQAKAAnAG0AcwBnACcALAAgACQAXwAuAEUAeABjAGUAcAB0AGkAbwBuAC4ATQBlAHMAcwBhAGcAZQApAAoAfQAKAEUAbABzAGUACgB7AAoAJABfAG8AYgBqAC4AQQBkAGQAKAAnAG0AcwBnACcALAAgACQAXwAuAFQAbwBTAHQAcgBpAG4AZwAoACkAKQAKAH0ACgBJAGYAIAAoACQAXwAuAEkAbgB2AG8AYwBhAHQAaQBvAG4ASQBuAGYAbwAuAFAAbwBzAGkAdABpAG8AbgBNAGUAcwBzAGEAZwBlACkACgB7AAoAJABfAG8AYgBqAC4AQQBkAGQAKAAnAGUAeABjAGUAcAB0AGkAbwBuACcALAAgACQAXwAuAEkAbgB2AG8AYwBhAHQAaQBvAG4ASQBuAGYAbwAuAFAAbwBzAGkAdABpAG8AbgBNAGUAcwBzAGEAZwBlACkACgB9AAoARQBsAHMAZQBJAGYAIAAoACQAXwAuAFMAYwByAGkAcAB0AFMAdABhAGMAawBUAHIAYQBjAGUAKQAKAHsACgAkAF8AbwBiAGoALgBBAGQAZAAoACcAZQB4AGMAZQBwAHQAaQBvAG4AJwAsACAAJABfAC4AUwBjAHIAaQBwAHQAUwB0AGEAYwBrAFQAcgBhAGMAZQApAAoAfQAKAFQAcgB5AAoAewAKACQAXwBvAGIAagAuAEEAZABkACgAJwBlAHIAcgBvAHIAXwByAGUAYwBvAHIAZAAnACwAIAAoACQAXwAgAHwAIABDAG8AbgB2AGUAcgB0AFQAbwAtAEoAcwBvAG4AIAB8ACAAQwBvAG4AdgBlAHIAdABGAHIAbwBtAC0ASgBzAG8AbgApACkACgB9AAoAQwBhAHQAYwBoAAoAewAKAH0ACgBFAGMAaABvACAAJABfAG8AYgBqACAAfAAgAEMAbwBuAHYAZQByAHQAVABvAC0ASgBzAG8AbgAgAC0AQwBvAG0AcAByAGUAcwBzACAALQBEAGUAcAB0AGgAIAA5ADkACgBFAHgAaQB0ACAAMQAKAH0ACgBGAGkAbgBhAGwAbAB5ACAAewAgAFIAZQBtAG8AdgBlAC0ASQB0AGUAbQAgACIAQwA6AFwAVQBzAGUAcgBzAFwAYQBuAHMAaQBiAGwAZQBcAEEAcABwAEQAYQB0AGEAXABMAG8AYwBhAGwAXABUAGUAbQBwAFwAYQBuAHMAaQBiAGwAZQAtAHQAbQBwAC0AMQA0ADcANwAzADEANgA2ADAAMAAuADcANwAtADEANgAzADIAMQA4ADYAOQAwADkANgA1ADkAOAA2ACIAIAAtAEYAbwByAGMAZQAgAC0AUgBlAGMAdQByAHMAZQAgAC0ARQByAHIAbwByAEEAYwB0AGkAbwBuACAAUwBpAGwAZQBuAHQAbAB5AEMAbwBuAHQAaQBuAHUAZQAgAH0A", "unreachable": true}
```

##### EXPECTED RESULTS

Playbook continues run.
##### ACTUAL RESULTS

Ansible connection is severed mid-play.

```

```

**Comment by ansibot** [comment]
@nitzmahone ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by nitzmahone** [comment]
@christonog What Windows flavor are you running? I've seen some OpenSSL oddities where the SSL tunnel gets wedged and causes a read timeout on 2008R2 (and have heard from others that it can happen on 2012R2, but never seen that one myself). Still not sure if the problem lies on the Windows side or the Linux side, though.

**Comment by ansibot** [comment]
@nitzmahone, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@nitzmahone, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by mindrunner** [comment]
We are facing exactly same problem here while provisioning Amazon AWS windows Hosts. (Windows_Server-2016-English-Full-Base-*)

**Comment by dagwieers** [comment]
Is this issue still a problem in v2.3 or the devel branch ? If so, or uncertain, please move this issue over to https://github.com/ansible/ansible so it is properly tracked. (I would like to clean up all open issues for Windows on this repository)

**Comment by mindrunner** [comment]
This issue still persists with Ansible 2.3. How can I move this issue?

**Comment by jhawkesworth** [comment]
@mindrunner - I think only the original issue creator can move issues.  However, if you are having similar problems, please raise a new issue on https://github.com/ansible/ansible with as much detail as possible.

**Comment by dagwieers** [comment]
The connection problems using WinRM that we experienced were related to intermittent network issues. Since we fixed the hardware-problem at the source of these network issues, WinRM has been quite stable.

In some cases (slow VPN) it can help to increase specific WinRM timeout settings.

**Comment by dagwieers** [comment]
So for future reference, there are two WinRM timeout parameters you can set:

```yaml
ansible_winrm_operation_timeout_sec: 120
ansible_winrm_read_timeout_sec: 150
```

I usually increase the `ansible_winrm_read_timeout_sec` parameter (from 30 to 90). Not sure if that is still needed, but it helped with the intermittent network issues.

Also see: https://github.com/ansible/ansible/issues/21797
Documentation updated.

**Comment by ntting** [comment]
@dagwieers "ansible_winrm_read_timeout_sec" is this parameter added? what's ansible version fixed this bug?

**Comment by jborean93** [comment]
Anything after ansible_winrm_ in your vars will be passed along to the pywinrm library, e.g. ansible_winrm_read_timeout_sec will pass along the variable read_timeout_sec to the init section of https://github.com/diyan/pywinrm/blob/master/winrm/protocol.py. If your pywinrm version installed supports that parameter then it will work.

As for the stability of WinRM, I've haven't come across any of these issues mentioned either where I work or locally in my home network. If your network is a bit temperamental it isn't very tolerant of dropped packets compared to SSH so like @dagwieers suggested you might want to look into your network.
